### PR TITLE
update-balance if link succesfully generated

### LINKED ISSUE #1636: 🐛 [Bug] Generating link does not update gdd balance
## 🐛 Bugreport

Generating link does not update gdd balance

1. Generate link Step 1 & 2 - remember balance
2. Have valid link in step 3
3. See balance is still the same in step 3
4. Refresh the balance (e.g. go to transaction) - have correct balance
5. ...
6. Profit

Please update the balance in step 3 of generating a link.

Might be linked to #1609

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
 die balance wird upgedated +  ~ decay für 14 tage
 
### Issues
 - fixes #1636 
 

https://user-images.githubusercontent.com/1324583/159435069-7c06aba9-09b2-4d89-8b41-70748835c365.mp4

### CODE REVIEW

File: frontend/src/pages/Send.vue [hunk 1]
```diff
@@ -143,6 +143,10 @@ export default {
             .then((result) => {
               this.code = result.data.createTransactionLink.code
               this.currentTransactionStep = TRANSACTION_STEPS.transactionResultLink
+              this.$emit(
+                'update-balance',
+                this.transactionData.amount + this.transactionData.amount * 0.028,
+              )
```
Comment (ulfgebhardt) [inline frontend/src/pages/Send.vue:149]: can we get this data from the backend?

File: frontend/src/components/GddSend/TransactionForm.vue [hunk 2]
```diff
@@ -160,15 +160,18 @@ export default {
   },
   props: {
     balance: { type: Number, default: 0 },
+    email: { type: String, default: '' },
```
Comment (Mogge) [inline frontend/src/components/GddSend/TransactionForm.vue:163]: I did this to restore the data, when the user cancels the confirmation to correct some input fields

File: frontend/src/components/GddSend/TransactionResultSendError.vue [hunk 3]
```diff
@@ -31,7 +31,7 @@
 </template>
 <script>
 export default {
-  name: 'TransactionResultSend',
+  name: 'TransactionResultSendError',
```
Comment (Mogge) [inline frontend/src/components/GddSend/TransactionResultSendError.vue:34]: Please always take care of correct naming!

File: frontend/src/pages/Send.vue [hunk 4]
```diff
@@ -3,7 +3,11 @@
     <b-container>
       <gdd-send :currentTransactionStep="currentTransactionStep" class="pt-3 ml-2 mr-2">
         <template #transactionForm>
-          <transaction-form :balance="balance" @set-transaction="setTransaction"></transaction-form>
+          <transaction-form
+            v-bind="transactionData"
```
Comment (Mogge) [inline frontend/src/pages/Send.vue:7]: to restore the data after canceling confirmation

File: frontend/src/pages/Send.spec.js [hunk 5]
```diff
@@ -48,8 +46,7 @@ describe('Send', () => {
       expect(wrapper.find('div.gdd-send').exists()).toBeTruthy()
```
Comment (ogerly) [inline frontend/src/pages/Send.spec.js:46]:
```suggestion
      expect(wrapper.find('div.gdd-send').exists()).toBe(true)
```

File: frontend/src/pages/Send.spec.js [hunk 6]
```diff
@@ -58,86 +55,92 @@ describe('Send', () => {
...
-        expect(wrapper.findComponent({ name: 'TransactionForm' }).exists()).toBeTruthy()
-        expect(wrapper.vm.transactionData).toEqual({
-          email: 'user@example.org',
-          amount: 23.45,
-          memo: 'Make the best of it!',
-          selected: SEND_TYPES.send,
-        })
-      })
+          it('shows the transaction formular again', () => {
+            expect(wrapper.findComponent({ name: 'TransactionForm' }).exists()).toBeTruthy()
```
Comment (ogerly) [inline frontend/src/pages/Send.spec.js:72]:
```suggestion
            expect(wrapper.findComponent({ name: 'TransactionForm' }).exists()).toBe(true)
```

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**